src/librustdoc/html/static/main.js

```diff
@@ -2913,6 +2913,7 @@ function defocusSearchBar() {
             ["&#9166;", "Go to active search result"],
             ["+", "Expand all sections"],
             ["-", "Collapse all sections"],
+            ["esc", "Close the search view"],
```

Comment:
It does a few other things like:

 * Removing the query parameters from the URL (like `?whatever=12`)
 * Closing the help popup
 * Closing the theme list if open
 * Blurring the focus from the search input